tests/test_models/test_mutables/test_derived_mutable.py

```diff
@@ -0,0 +1,188 @@
+            num_channels=3,
+            candidate_choices=[1, 2, 3],
+            candidate_mode='number')
+        mc_derived = mc * 3
+        assert mc_derived.source_mutables == {mc}
+
+        mc.current_choice = 1
+        assert mc_derived.current_choice == 3
+        assert torch.equal(
+            mc_derived.current_mask,
+            torch.tensor([1, 1, 1, 0, 0, 0, 0, 0, 0], dtype=torch.bool))
+
+        mc.current_choice = 2
+        assert mc_derived.current_choice == 6
+        assert torch.equal(
+            mc_derived.current_mask,
+            torch.tensor([1, 1, 1, 1, 1, 1, 0, 0, 0], dtype=torch.bool))
+
+    def test_mutable_divide(self) -> None:
+        mc = OneShotMutableChannel(
+            num_channels=128,
+            candidate_choices=[112, 120, 128],
+            candidate_mode='number')
+        mc_derived = mc // 8
+        assert mc_derived.source_mutables == {mc}
+
+        mc.current_choice = 128
+        assert mc_derived.current_choice == 16
+        assert torch.equal(mc_derived.current_mask,
+                           torch.ones(16, dtype=torch.bool))
+        mc.current_choice = 120
+        assert mc_derived.current_choice == 16
+        assert torch.equal(mc_derived.current_mask,
+                           torch.ones(16, dtype=torch.bool))
+
+        mv = OneShotMutableValue(value_list=[112, 120, 128])
+        mv_derived = mv // 8
+        assert mv_derived.source_mutables == {mv}
+
+        mv.current_choice == 128
+        assert mv_derived.current_choice == 16
+        mv.current_choice == 120
+        assert mv_derived.current_choice == 16
+
+    def test_double_fixed(self) -> None:
+        choice_fn = lambda x: x  # noqa: E731
+        derived_mutable = DerivedMutable(choice_fn)
+        derived_mutable.fix_chosen({})
+
+        with pytest.raises(RuntimeError):
+            derived_mutable.fix_chosen({})
+
+    def test_source_mutables(self) -> None:
+        mc1 = OneShotMutableChannel(
+            num_channels=3, candidate_choices=[1, 3], candidate_mode='number')
+        mc2 = OneShotMutableChannel(
+            num_channels=4, candidate_choices=[1, 4], candidate_mode='number')
+        ms = [mc1, mc2]
+
+        mc_derived1 = DerivedMutable.derive_concat_mutable(ms)
+
+        from mmrazor.models.mutables.derived_mutable import (_concat_choice_fn,
+                                                             _concat_mask_fn)
+        mc_derived2 = DerivedMutable(
+            choice_fn=_concat_choice_fn(ms),
+            mask_fn=_concat_mask_fn(ms),
+            source_mutables=ms)
+        assert mc_derived1.source_mutables == mc_derived2.source_mutables
+
+        dd_mutable = mc_derived1.derive_same_mutable()
+        assert dd_mutable.source_mutables == mc_derived1.source_mutables
+
+        with pytest.raises(ValueError):
+            _ = DerivedMutable(
+                choice_fn=lambda x: x, source_mutables=[mc_derived1])
+
+        def dict_closure_fn(x, y):
```

Comment:
Add a UT for the case when there are source_mutables and/or derived_mutables in the closure function.

Reply:
I am not sure can [this UT](https://github.com/open-mmlab/mmrazor/pull/215/files#diff-c697a0e22a3d96de082519250e753c5376f3ff7d96558822fbcb5c9b4dad4aebR157) meet the requirement.